### PR TITLE
Add candidate rejected email for when offers have been made

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

When a candidate receives a rejection:

- if they have one or multiple offers 
- and are not awaiting any additional responses 

They should receive an email informing them of the rejection and their next steps.

This is the fourth and final PR on this ticket (previous ones are #1358, #1371  and #1378 )

## Changes proposed in this pull request

- Adds an additional email (application_rejected_offers_made)
- Updates the service to send this email

## Guidance to review

This effectively combines the following two emails into one: 

-You have 1 new rejection AND there are no outstanding provider decisions AND you have 2 or more offers(to a candidate)
- You have 1 new rejection AND there are no outstanding provider decisions AND you only have 1 offer (to a candidate)

I chose to do this as they're fairly similar. Does that seem reasonable?

https://docs.google.com/document/d/1VH_nxuLTiVCkmKhC4eaBpFC61Hd5_Ol1tKwOul-2uUs/edit#

## Link to Trello card

https://trello.com/c/22e7C80p/840-email-%F0%9F%99%85%E2%99%80%EF%B8%8F-a-provider-has-rejected-your-application-to-candidate

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/views/candidate_mailer/application_rejected/offers_made.erb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# Make a decision about your <%= 'offer'.pluralize(@application.offers.count) %> by <%= @application.decline_by_default_at %>
+
+<% if @application.offers.count > 1  %>
+
+You’ve received the following offers:
+<% @application.offers.each do |offer| %>
+  - <%= "#{offer.course.name_and_code} at #{offer.provider.name}"%>
+<% end %>
+
+If you don’t reply within 10 working days, your applications will be withdrawn.
```
Comment (benilovj) [inline app/views/candidate_mailer/application_rejected/offers_made.erb]: The `10` should not be hard-coded – we should be using the `decline by default` duration out of the time limit config.

File: app/views/candidate_mailer/application_rejected/offers_made.erb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+<% else %>
+
+You now have 10 working days to make a decision about your offer for a place on <%= @application.offers.first.course.name_and_code %> at <%= @application.offers.first.provider.name%>.
```
Comment (benilovj) [inline app/views/candidate_mailer/application_rejected/offers_made.erb]: as above

File: app/views/candidate_mailer/new_offer/single_offer.text.erb [hunk 3]
```diff
@@ -14,7 +14,7 @@ Contact <%= @provider_name %> directly if you have any questions about this.
 
 # Make a decision by <%= @application_choice.decline_by_default_at&.to_s(:govuk_date) %>
 
-You have 10 working days to make a decision. If you don’t reply by <%= @application_choice.decline_by_default_at&.to_s(:govuk_date) %>, your application will be withdrawn. 
+You have 10 working days to make a decision. If you don’t reply by <%= @application_choice.decline_by_default_at&.to_s(:govuk_date) %>, your application will be withdrawn.
```
Comment (benilovj) [inline app/views/candidate_mailer/new_offer/single_offer.text.erb]: as above

File: app/views/candidate_mailer/application_rejected/offers_made.erb (hunk 1 above)
Comment (davidgisbey) [reply app/views/candidate_mailer/application_rejected/offers_made.erb]: Thanks

File: config/locales/application_choice_rejected_email.yml [hunk 4]
```diff
@@ -3,6 +3,7 @@ en:
     subject:
       all_rejected: "%{provider_name} has responded: next steps"
       awaiting_decisions: "%{provider_name} has made a decision on your application for %{course_name}"
+      offers_made: "%{provider_name} has responded: make a decision within 10 working days"
```
Comment (tvararu) [inline config/locales/application_choice_rejected_email.yml]: Should this `10` also be based on `TimeLimitConfig`?

File: spec/system/candidate_interface/candidate_receives_rejection_email_spec.rb [hunk 5]
```diff
@@ -14,6 +14,16 @@
     when_i_am_awaiting_decisions_and_have_no_offers
     and_a_provider_rejects_my_application
     then_i_receive_the_application_rejected_awaiting_decisions_email
+
+    when_i_have_a_single_offer
+    and_a_provider_rejects_my_application
+    then_i_receive_the_application_rejected_offers_made_email
+    and_it_includes_details_of_my_offer
+
+    when_i_am_have_multiple_offers
```
Comment (tvararu) [inline spec/system/candidate_interface/candidate_receives_rejection_email_spec.rb]:
```suggestion
    when_i_have_multiple_offers
```

File: spec/system/candidate_interface/candidate_receives_rejection_email_spec.rb [hunk 6]
```diff
@@ -36,6 +46,19 @@ def when_i_am_awaiting_decisions_and_have_no_offers
     @application_choice = @application_form.application_choices.first
   end
 
+  def when_i_have_a_single_offer
+    @application_form = create(:completed_application_form)
+    @offer = create(:application_choice, status: :offer, application_form: @application_form, decline_by_default_at: 10.business_days.from_now)
+    @application_choice = create(:application_choice, status: :awaiting_provider_decision, application_form: @application_form)
+  end
+
+  def when_i_am_have_multiple_offers
```
Comment (tvararu) [inline spec/system/candidate_interface/candidate_receives_rejection_email_spec.rb]:
```suggestion
  def when_i_have_multiple_offers
```

File: app/mailers/candidate_mailer.rb [hunk 7]
```diff
@@ -91,21 +95,30 @@ def application_rejected(application_choice, template_name)
     decisions = application_choice.application_form.application_choices.select(&:awaiting_provider_decision?).map do |decision|
       "#{decision.course_option.course.name_and_code} at #{decision.course_option.course.provider.name}"
     end
+    offers = application_choice.application_form.application_choices.select(&:offer?)
+    decline_by_default_at = offers.map(&:decline_by_default_at).compact.max&.to_s(:govuk_date)
+    dbd_days = offers.map(&:decline_by_default_days).max
```
Comment (tijmenb) [inline app/mailers/candidate_mailer.rb]: This seems to be nil, causing an empty thing in the subject.

File: app/views/candidate_mailer/application_rejected/offers_made.erb [hunk 8]
```diff
@@ -0,0 +1,40 @@
+Dear <%= @application.candidate_name %>,
+
+# Application decision
+
+<%= @application.provider_name %> has decided not to progress your teacher training application for <%= @application.course_name %> on this occasion. They gave the following feedback:
+
+<%= @application.rejection_reason %>
```
Comment (tijmenb) [inline app/views/candidate_mailer/application_rejected/offers_made.erb]: Make this a blockquote?

```suggestion
> <%= @application.rejection_reason %>
```

File: app/mailers/candidate_mailer.rb [hunk 9]
```diff
@@ -73,6 +73,10 @@ def application_rejected_awaiting_decisions(application_choice)
     application_rejected(application_choice, :awaiting_decisions)
   end
 
+  def application_rejected_offers_made(application_choice)
```
Comment (fofr) [inline app/mailers/candidate_mailer.rb]: I'm wondering if these methods should be a little clearer, it took me a short while to work out what was going on here.
Maybe: `application_rejected_but_with_other_offers_made`

Comment (tijmenb) [reply app/mailers/candidate_mailer.rb]: We're going to do a follow up PR to refactor the previous PRs, we'll take this onboard.

**Comment by tijmenb** [comment]
Merged to get started on some refactoring of the CandidateMailer.

**Pull request merged**